homeassistant/components/prosegur/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The Prosegur Alarm integration."""
+import logging
+
+from pyprosegur.auth import Auth
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_COUNTRY, DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Prosegur Alarm from a config entry."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        hass.data.setdefault(DOMAIN, {})
+        hass.data[DOMAIN][entry.entry_id] = Auth(
+            session,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_COUNTRY],
+        )
+        await hass.data[DOMAIN][entry.entry_id].login()
+
+    except ConnectionRefusedError as error:
+        _LOGGER.error("Configured credential are invalid, %s", error)
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
```

Comment:
Nowadays we just raise `ConfigEntryAuthFailed` and it will start the reauth flow automatically.

Reply:
I actually had this before the suggestion from https://github.com/home-assistant/core/pull/44679#discussion_r675989110 :) but no problem